M2/Macaulay2/packages/Macaulay2Doc/functions/replace-doc.m2

```diff
@@ -21,29 +21,40 @@ doc ///
     re:String
       a @TO2 {"regular expressions", "regular expression"}@ describing a pattern
     replacement:String
-      a replacement or regex formatting string, may include backreferences
+      following the @TO2 {"regular expressions", "formatting syntax"}@
```

Comment:
I had it that way at first, but the end result of your suggestion is somewhat jarring:
> replacement, a string, a replacement string, following the formatting syntax

Currently it is this, which I think is more readable and carries all the information:
> replacement, a string, following the formatting syntax

Don't you agree that the former is redundant?

Reply:
Also, my usage matches the one in the `simpleDocFrob` example:
![image](https://user-images.githubusercontent.com/3147718/91769968-f6256b00-eba5-11ea-8a92-299790290a37.png)

Reply:
> I had it that way at first, but the end result of your suggestion is somewhat jarring:
> 
> > replacement, a string, a replacement string, following the formatting syntax
> 
> Currently it is this, which I think is more readable and carries all the information:
> 
> > replacement, a string, following the formatting syntax
> 
> Don't you agree that the former is redundant?

No, because the identifier being described, "replacement", is not part of its own description, so the descriptive text your way contains only "a string, following the formatting syntax", is is both non-grammatical, because of the comma, and incomplete.

Reply:
I don't think it's non-grammatical to say, for instance, "X, a Sequence, following the description below". So why is it non-grammatical? They are not supposed to be full sentences. Indeed, the "conventions for documentation" says:
> Inputs, Outputs, and Consequences should not end with periods

Reply:
Here is a full sentence: "The inputs of this function are: X, a Sequence, following the description above; Y, a String, at most 5 characters; Z, an Integer, positive." Is there a grammatical error here?

Reply:
What's incomplete about  "a string, following the formatting syntax"  is that it doesn't say what the string is used for.

Reply:
> What's incomplete about "a string, following the formatting syntax" is that it doesn't say what the string is used for.

It doesn't have to, because the description of the inputs only needs to clarify what the input can be, not what it is going to be used for. That's the job of other parts. In particular, the description of the output clarifies it:
> a string, obtained from str by replacing its substrings matching re by copies of replacement

Reply:
> > What's incomplete about "a string, following the formatting syntax" is that it doesn't say what the string is used for.
> 
> It doesn't have to, because the description of the inputs only needs to clarify what the input can be, not what it is going to be used for. That's the job of other parts. 

That's not the way I think of it.  Oh, well, to each his own.

Reply:
Shall I mark this, and the other "antecedent missing" comments, as resolved?

Reply:
Sure.